internal/sql/migrations.go

```diff
@@ -27,7 +25,7 @@ type (
 
 	MainMigrator interface {
 		Migrator
-		MakeDirsForPath(ctx context.Context, tx Tx, path string) (int64, error)
+		InsertDirectoriesMemoized(ctx context.Context, tx Tx, bucketID int64, path string, memo map[string]int64) (int64, error)
```

Comment:
I did consider it but wanted to avoid calculating the paths twice. Just considered it a viable tradeoff to have it in the `MainMigrator` interface but not the main database tx interface. I'll change it though I wasn't too content with it either.